tests/tests/test_security.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python
+# Copyright 2016 Mender Software AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from fabric.api import *
+import pytest
+import time
+from common import *
+from common_docker import *
+from common_setup import *
+from helpers import Helpers
+from MenderAPI import auth, adm, deploy, image, logger
+from common_update import common_update_procedure
+from mendertesting import MenderTesting
+import subprocess
+import sys
+sys.path.insert(0, '..')
+import conftest
+
+
+class TestSecurity(MenderTesting):
+
+    @pytest.mark.usefixtures("standard_setup_without_client")
+    def test_ssl_only(self):
+        """ make sure we are not exposing any non-ssl connections"""
+
+        # get all exposed ports from docker
+        exposed_hosts = subprocess.check_output("docker ps | grep %s | grep -o -E '0.0.0.0:[0-9]*'" % (conftest.docker_compose_instance), shell=True)
+        print exposed_hosts.split()
+
+        for host in exposed_hosts.split():
```

Comment:
Works as long as we're exposing http over SSL which may not be the case in the future. IMO a better way would be to check if the port is secured like this:

```
import ssl
import socket
import contextlib

for host in exposed_hosts.split():
    with contextlib.closing(ssl.wrap_socket(socket.socket())) as sock:
       with pytest.raises(ssl.SSLError):
           sock.connect()
```    
on top of this you can force specific versions of the protocol, ex. `ssl.wrap_socket(socket.socket(), ssl_version=ssl.PROTOCOL_TLS)` and so on (NOTE: ymmv with python 2.x)